src/signals/incident-management/containers/IncidentDetail/components/AttachmentViewer/index.js

```diff
@@ -39,39 +39,73 @@ const Img = styled.img`
   margin: 0 auto;
 `;
 
-const AttachmentViewer = ({ href, attachments, onShowAttachment }) => {
-  const index = attachments.findIndex(item => item.location === href);
+const AttachmentViewer = ({ href, attachments }) => {
+  const [currentHref, setCurrentHref] = useState(href);
+  const index = attachments.findIndex(item => item.location === currentHref);
   const previous = index > 0 ? attachments[index - 1].location : false;
   const next = index < attachments.length - 1 ? attachments[index + 1].location : false;
 
+  const handleKeyDown = useCallback(
```

Comment:
Shouldn't this only work when this control has the focus? Now it would react when it is rendered and could interfere with any other control that uses keyboard events.

Reply:
Good point, I'll check what happens when the attachment viewer is open and the focus is within an input element.

Reply:
Adjusted. Thanks for pointing the issue out.